File: 프로그래머스/lv1/42840. 모의고사/모의고사.java
import java.util.*;
class Solution {
    public int calculate(int[] answers, int[] pattern) {
        int cnt = 0;

        for (int i = 0; i < answers.length; i++) {
            if(answers[i] == pattern[i % pattern.length]) {
                cnt++;
            }
        }
        return cnt;
    }

    public int[] solution(int[] answers) {

        int[][] pattern = {{1, 2, 3, 4, 5},{2, 1, 2, 3, 2, 4, 2, 5},{3, 3, 1, 1, 2, 2, 4, 4, 5, 5}};
        int cnt[] = new int[3];
        for (int i = 0; i < pattern.length; i++) {
            cnt[i] = calculate(answers,pattern[i]);
        }
        
        
        int maxcnt = Math.max(cnt[0],Math.max(cnt[1],cnt[2]));

        ArrayList<Integer> list = new ArrayList<>();
        for (int i = 0; i < cnt.length; i++) {
            if(maxcnt == cnt[i]) list.add(i+1);
        }

        int[] answer = new int[list.size()];
        for (int i = 0; i < list.size(); i++) {
            answer[i] = list.get(i);
        }

        return answer;
    }
}

File: 프로그래머스/lv1/42840. 모의고사/모의고사.py
def solution(answers):

    answer = []
    a = [1,2,3,4,5]
    b = [2,1,2,3,2,4,2,5]
    c = [3,3,1,1,2,2,4,4,5,5]
    records = [0 for _ in range(3)]
    
    for i in range(len(answers)):
        if a[i%5] == answers[i]:
            records[0] += 1
    for i in range(len(answers)):
        if b[i%8] == answers[i]:
            records[1] += 1
    for i in range(len(answers)):
        if c[i%10] == answers[i]:     
            records[2] += 1
        

    for i in range(3):
        if records[i] == max(records):
            answer.append(i+1)
        
    return answer